### PR TITLE
Fix output buffer is imcomplete working in child_process

### PR DESCRIPTION
**Pull request by ysugimoto**
When I send a large buffer, over 50KB's CSS string, working in `child_process.spawn`, `node-sass` outputs only partial strings.

Certainly, it is caused by nodejs's `console.log` max size.
https://github.com/nodejs/node/issues/9546

So I modified to send output by continuous partial strings, its asynchronous.

This PR will fix this issue: https://github.com/sass/node-sass/issues/1823.

### CODE REVIEW

File: bin/node-sass [hunk 1]
```diff
@@ -161,11 +163,17 @@ function getEmitter() {
...
+    asyncWaiting = true;
+
+    // In nodejs <= 7.1.0, console.log cannot write output larger than 8275 bytes data.
+    // So that, we output partial strings manually with some asynchronous way.
+    stdout(data);
   });
 
   emitter.on('done', function() {
-    if (!options.watch && !options.directory) {
+    if (!options.watch && !options.directory && !asyncWaiting) {
```
Comment (xzyfer) [inline bin/node-sass]: isn't `asyncWaiting` always true now?

**Comment by xzyfer** [comment]
https://www.npmjs.com/package/stdout-stream seems like a better alternative than writing out own stdout stream lib.

File: bin/node-sass (hunk 1 above)
Comment (ysugimoto) [inline bin/node-sass]: `asyncWaiting` is initialized on false when called `getEmitter()`.
https://github.com/sass/node-sass/pull/1834/files#diff-c19ab2481e2703fcf0d2808f4c762f63R147

I read project source code, emit `log` event only stdout case ( is it right? ).
So that, `asyncWaiting` set to be true only emitted `log` event.

Please advise if there are other concerns that you need to worry about. thanks!

**Comment by ysugimoto** [comment]
I replaced out stdout lib to suggested package, it also works fine!

**Comment by ysugimoto** [comment]
It works fine, but Travis CI is down for some testings...

File: bin/node-sass (hunk 1 above)
Comment (xzyfer) [inline bin/node-sass]: I don't think this is right. I believe we always want to exit when emitter.log is called. This is what happens when you call the node-sass cli. If I understand correctly all we need to do is

```js
emitter.on('log', stdout.write);
```

Am I missing something?

**Comment by xzyfer** [comment]
I'm sorry the issue here is not clear. 

Either the problem is related to `console.log`s max buffer size, in which we can fix by

```
emitter.on('log', stdout.write)
```

It's because `process.stdout` is no longer blocking, in which case we can use https://github.com/isaacs/block-stream. Either way i don't think this is the correct solution.

**Comment by ysugimoto** [comment]
Yes, I think so.

However, `process.stdout` is no longer blocking, then we stop `process.exit` until stdout stream output completely. It's called by `emitter.emit('done')` immediately, So I add the flag of `asyncWaiting`.

**Comment by xzyfer** [comment]
OK I did some digging. You're correct that the issue is due how [process.exit](https://nodejs.org/api/process.html#process_process_exit_code) works.

> It is important to note that calling `process.exit()` will force the process to exit as quickly as possible _even if there are still asynchronous operations pending_ that have not yet completed fully, _including_ I/O operations to `process.stdout` and `process.stderr`.

I see ways to solve this:
- remove the `process.exit` (I don't think we need it)
- use [exit](https://www.npmjs.com/package/exit) instead

**Comment by ysugimoto** [comment]
Thank you for your advice. I replaced to `process.exit` to `exit` package.

And I removed exiting in `emitter.on('done', ...)` callback because of process exited even though using `stdout-stream` package, I wondering...
After removed this point, It works fine: CLI, spawned process, and watch mode.

File: bin/node-sass [hunk 2]
```diff
@@ -161,13 +163,12 @@ function getEmitter() {
   });
 
   emitter.on('log', function(data) {
-    console.log(data);
+    // In nodejs <= 7.1.0, console.log cannot write output larger than 8275 bytes data.
+    stdout.write(data);
   });
 
   emitter.on('done', function() {
```
Comment (xzyfer) [inline bin/node-sass]: Please delete the `emitter.on('done'` block all together.

**Comment by ysugimoto** [comment]
https://github.com/sass/node-sass/pull/1834/files#diff-c19ab2481e2703fcf0d2808f4c762f63R165

Use `bind()` because of avoid error on `stdout-stream`.

File: package.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
     "async-foreach": "^0.1.3",
     "chalk": "^1.1.1",
     "cross-spawn": "^3.0.0",
+    "exit": "^0.1.2",
```
Comment (xzyfer) [inline package.json]: Please remove.

File: lib/render.js [hunk 4]
```diff
@@ -47,21 +47,14 @@ module.exports = function(options, emitter) {
   var stdin = options.stdin;
 
   var success = function(result) {
-    var todo = 1;
```
Comment (xzyfer) [inline lib/render.js]: The `done` still needs to be fired. It's used in other places. Please undo these changes and just remove 

```
emitter.on('done', function() {		
    // noop		
});
```

Comment (xzyfer) [inline lib/render.js]: The `done` still needs to be fired. It's used in other places. Please undo these changes and just remove 

```
emitter.on('done', function() {		
    // noop		
});
```

**Comment by xzyfer** [comment]
Thank you for your work and patience. I'll merge this when CI is happy.
